plugins/repository-gcs/src/main/java/io/crate/gcs/GCSClientSettings.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package io.crate.gcs;
+
+import java.net.URI;
+
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+
+import com.google.auth.oauth2.ServiceAccountCredentials;
+
+import io.crate.common.unit.TimeValue;
+import io.crate.exceptions.InvalidArgumentException;
+
+
+public class GCSClientSettings {
+
+    static final Setting<SecureString> PRIVATE_KEY_ID_SETTING = Setting.maskedString("private_key_id");
+
+    static final Setting<SecureString> PRIVATE_KEY_SETTING = Setting.maskedString("private_key");
+
+    static final Setting<SecureString> CLIENT_EMAIL_SETTING = Setting.maskedString("client_email");
+
+    static final Setting<SecureString> CLIENT_ID_SETTING = Setting.maskedString("client_id");
+
+    static final Setting<String> ENDPOINT_SETTING = Setting.simpleString("endpoint", Setting.Property.NodeScope);
+
+    static final Setting<String> PROJECT_ID_SETTING = Setting.simpleString("project_id", Setting.Property.NodeScope);
+
+    static final Setting<String> TOKEN_URI_SETTING = Setting.simpleString(
+        "token_uri", "https://oauth2.googleapis.com/token", Setting.Property.NodeScope);
+
+    static final Setting<TimeValue> CONNECT_TIMEOUT_SETTING = Setting.timeSetting(
+        "connect_timeout", TimeValue.timeValueSeconds(20), TimeValue.MINUS_ONE, Setting.Property.NodeScope);
+
+    static final Setting<TimeValue> READ_TIMEOUT_SETTING = Setting.timeSetting(
+        "read_timeout", TimeValue.timeValueSeconds(20), TimeValue.MINUS_ONE, Setting.Property.NodeScope);
```

Comment:
shouldn't be in NodeScope?

Reply:
All settings from plugins are required to be scoped otherwise SettingsModule#registerSetting rejects the setting.

Reply:
I would do this in a follow-up. Currently, the NodeScope for the Settings is needed for plugins and is done the same way in all other plugins. We can change the behaviour, but this is not the scope of this pr.

Reply:
@seut I had a quick look into the underlying logic. All Settings are grouped into IndexScopedSettings or ClusterSettings. Where would you expect that a Setting without a Scope defined would grouped in? 
See https://github.com/crate/crate/blob/master/server/src/main/java/org/elasticsearch/common/settings/SettingsModule.java for details.